homework04/frontend/components/RequestVotingTokens.tsx

```diff
@@ -21,13 +25,32 @@ export default function RequestVotingTokens(params: {address: `0x${string}` | un
   }
 
   if (isLoading) return <div>Requesting tokens from API...</div>;
-  if (!data) return (
-    <button
-      className="rounded-full bg-indigo-600 px-2.5 py-1 text-xs font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
-      onClick={handleRequestVotingTokens}>
-        Mint Tokens
-    </button>
-  );
+  if (!data) {
+    return (
+      <div className="flex flex-col justify-center">
+        <form className="flex flex-col justify-center mt-3">
+          <label>
+            Enter amount of voting token you want to request:
+          </label>
+          <input
+            className="mt-2"
+            type="number"
+            min={0.1}
+            max={1.0}
+            step={0.1}
+            placeholder="1"
+            value={voteAmount}
+            onChange={(e) => setVoteAmount(e.target.value)}
+          />
+        </form>
+        <button
+          className="mt-2 rounded-full bg-indigo-600 px-4 py-2 text-sm font-normal text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
+          onClick={handleRequestVotingTokens}>
+            Request Voting Tokens
```

Comment:
I'll change the input parameter to specify the amount to be minted instead of address, then